userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -207,20 +207,35 @@ bool cri_async_source::lookup_sync(const libsinsp::cgroup_limits::cgroup_limits_
 
 cri::cri(container_cache_interface &cache) : container_engine_base(cache)
 {
-	if(s_cri_unix_socket_path.empty()) {
-		return;
-	}
-
-	auto cri_path = scap_get_host_root() + s_cri_unix_socket_path;
-	struct stat s = {};
-	if(stat(cri_path.c_str(), &s) != 0 || (s.st_mode & S_IFMT) != S_IFSOCK) {
-		return;
+	if (s_cri_unix_socket_paths.empty())
+	{
+		// Default value when empty
+		s_cri_unix_socket_paths.emplace_back("/run/containerd/containerd.sock");
 	}
 
-	m_cri = std::unique_ptr<libsinsp::cri::cri_interface>(new libsinsp::cri::cri_interface(cri_path));
-	if(!m_cri->is_ok())
+	// Try all specified unix socket paths
+	for (auto &p : s_cri_unix_socket_paths)
 	{
-		m_cri.reset(nullptr);
+		if(p.empty()) {
+			continue;
+		}
+
+		auto cri_path = scap_get_host_root() + p;
+		struct stat s = {};
+		if(stat(cri_path.c_str(), &s) != 0 || (s.st_mode & S_IFMT) != S_IFSOCK)
+		{
+			continue;
+		}
+
+		m_cri = std::unique_ptr<libsinsp::cri::cri_interface>(new libsinsp::cri::cri_interface(cri_path));
+		if(!m_cri->is_ok())
+		{
+			m_cri.reset(nullptr);
```

Comment:
Maybe I'm not seeing the bigger picture, but don't we risk adding the same CRI socket path twice in this loop?

Reply:
Mmh i don't get it: as soon as a working socket is found, we use it and break out of the loop!
Am i missing anything obvious?

Reply:
Yeah maybe a comment to make this explicit would be good. Either way, I agree this is definitely something we can avoid supporting for now.

Reply:
Yeah, the rationale is that having multiple container runtimes on the same host is a sporadic case, so we wouldn't make things complex to support that. On the other hand, specifying the `--cri` multiple times (and using only the first match) will solve the "same config, multiple hosts" use case.

Reply:
Should i document it somewhere in this PR?